frontend/packages/dev-console/src/components/topology/TopologyEdgePanel.tsx

```diff
@@ -36,7 +32,11 @@ const connectorTypeToTitle = (type: string): string => {
   }
 };
 
-const TopologyEdgePanel: React.FC<TopologyEdgePanelProps> = ({ edge, consoleLinks }) => {
+const TopologyEdgePanel: React.FC<TopologyEdgePanelProps> = ({ edge }) => {
+  const [consoleLinks] = useK8sWatchResource<K8sResourceKind[]>({
```

Comment:
I'm a little hesitant to open a watch per topology node, even though there should only be one watch. It would be better if we can watch this once in the topology view.

This seems to be used for the Kiali link, but the logic looks incorrect here:

https://github.com/spadgett/console/blob/e47665c50e6f5ea842a2de42e88e60a3c2090452/frontend/packages/dev-console/src/components/topology/topology-utils.ts#L59-L61

It's assuming any namespace link is Kiali, but that's not necessarily the case. Maybe we can add a label to the Kiali `ConsoleLink` so we know.

cc @christianvogt

Reply:
Waiting on input from @christianvogt (who I believe is out on PTO).

Reply:
@spadgett why can't we use `useK8sWatchResource` multiple times on the page with the same resource?  i thought we can trust the framework to manage only a single watch when using the hook multiple times on the same page. 

I ask because I want to move in this direction because it makes it easier for each node to manage the data it needs instead of always needing to surface this information to the topology root.

Reply:
@christianvogt I was worried that we might have a race if two components start watching at the same time, but I think we're probably OK. The code to handle it is here:

https://github.com/spadgett/console/blob/b6ce43d2a4247538016b88e565288770feebe016/frontend/public/actions/k8s.ts#L146-L150

We should test to make sure only one list request and WebSocket gets opened.

Reply:
> We should test to make sure only one list request and WebSocket gets opened.

@christianvogt, can you either do said testing or help me understand what I need to do to view `<TopologyEdgePanel>`?

Reply:
`TopologyEdgePanel` is only rendered when selecting a link. Therefore there will only be a single watch here.

As for watching per workload node, I saw no extra network calls for the console links since a socket was already opened. Switching from and to topology multiple times with several workloads presents shows no additional console link watches.

Reply:
@spadgett, are you therefore good with this PR as is?